transactions/src/main/java/org/ehcache/transactions/xa/XAStore.java

```diff
@@ -0,0 +1,908 @@
+            }
+            if (o1.getValue().getTransactionId() == null && o2.getValue().getTransactionId() == null) {
+              return 0;
+            }
+            return o1.getValue().getTransactionId() != null ? 1 : -1;
+          }
+        };
+
+        // expiry
+        final Expiry<? super K, ? super V> configuredExpiry = storeConfig.getExpiry();
+        Expiry<? super K, ? super SoftLock> expiry = new Expiry<K, SoftLock>() {
+          @Override
+          public Duration getExpiryForCreation(K key, SoftLock softLock) {
+            if (softLock.getTransactionId() != null) {
+              // phase 1 prepare, create -> forever
+              return Duration.FOREVER;
+            } else {
+              // phase 2 commit, or during a TX's lifetime, create -> some time
+              return configuredExpiry.getExpiryForCreation(key, (V) softLock.getOldValue());
+            }
+          }
+
+          @Override
+          public Duration getExpiryForAccess(K key, SoftLock softLock) {
+            if (softLock.getTransactionId() != null) {
+              // phase 1 prepare, access -> forever
+              return Duration.FOREVER;
+            } else {
+              // phase 2 commit, or during a TX's lifetime, access -> some time
+              return configuredExpiry.getExpiryForAccess(key, (V) softLock.getOldValue());
+            }
+          }
+
+          @Override
+          public Duration getExpiryForUpdate(K key, SoftLock oldSoftLock, SoftLock newSoftLock) {
+            if (oldSoftLock.getTransactionId() == null) {
+              // phase 1 prepare, update -> forever
+              return Duration.FOREVER;
+            } else {
+              // phase 2 commit, or during a TX's lifetime
+              if (oldSoftLock.getOldValue() == null) {
+                // there is no old value -> it's a CREATE, update -> create -> some time
+                return configuredExpiry.getExpiryForCreation(key, (V) oldSoftLock.getOldValue());
+              } else {
+                // there is an old value -> it's an UPDATE, update -> some time
+                return configuredExpiry.getExpiryForUpdate(key, (V) oldSoftLock.getOldValue(), (V) oldSoftLock.getNewValueHolder().value());
+              }
+            }
+          }
+        };
+
+        // check if the store is persistent, and get the PersistenceSpaceIdentifier if it is
+        LocalPersistenceService.PersistenceSpaceIdentifier persistenceSpaceId = null;
+        ResourcePools resourcePools = storeConfig.getResourcePools();
+        for (ResourceType resourceType : resourcePools.getResourceTypeSet()) {
+          ResourcePool pool = resourcePools.getPoolForResource(resourceType);
+          if (pool.isPersistent()) {
+            try {
+              LocalPersistenceService persistenceService = serviceProvider.getService(LocalPersistenceService.class);
+              persistenceSpaceId = persistenceService.getOrCreatePersistenceSpace(uniqueXAResourceId);
```

Comment:
Our understandings aren't aligned: mine is that this code creates a subspace into the cache's space. I'll investigate and if I'm wrong, I'll change this code accordingly.